src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx

```diff
@@ -64,16 +68,20 @@ export const KeyValueTable = <K extends string>({
 
     // STATES
     const [updatedRows, setUpdatedRows] = useState<KeyValueRow<K>[]>(rows)
+    /** State to trigger useEffect to trigger autoFocus */
     const [newRowAdded, setNewRowAdded] = useState(false)
 
+    const isActionDisabled = readOnly || isAdditionNotAllowed
+
     /** Boolean determining if table has rows. */
     const hasRows = (!readOnly && !isAdditionNotAllowed) || !!updatedRows.length
+    const isFirstRowEmpty = !updatedRows[0]?.data[firstHeaderKey].value && !updatedRows[0]?.data[secondHeaderKey].value
```

Comment:
@RohitRaj011 Pls confirm if the data[firstHeaderKey] can be null

Reply:
can't be null